cmd/chart/upgrade_test.go

```diff
@@ -0,0 +1,97 @@
+package chart
+
+import (
+	"os"
+	"testing"
+
+	"gopkg.in/yaml.v2"
+)
+
+type test struct {
+	name     string
+	image    string
+	expected string
+}
+
+func Test_ChartUpgrade(t *testing.T) {
+	tests := []test{
+		{
+			name:     "builder_v11_rootless",
+			image:    "moby/buildkit:v0.11.5-rootless",
+			expected: "moby/buildkit:v0.11.6-rootless",
+		},
+		{
+			name:     "builder_v10_rootless",
+			image:    "moby/buildkit:v0.10.0-rootless",
+			expected: "moby/buildkit:v0.11.6-rootless",
+		},
+		{
+			name:     "builder_v8_no_rootless",
+			image:    "moby/buildkit:v0.8.0",
+			expected: "moby/buildkit:v0.11.6",
+		},
+		{
+			name:     "prom_Stable",
```

Comment:
I saw you excluded these unit tests..

Why? Is it because they're integration tests which actually call out to the Internet?

How about providing a canned list of available tags here in a similar format to what crane returns.

See the unit testing section of my eBook that I gifted to you.

You should see a pattern for stubbing out data.

Reply:
```
name: "Promote from a stable tag to the next stable tag"
current: "prom/prometheus:v2.42.0"
want: "prom/prometheus:v2.45.0"
tags: []{ "prom/prometheus:v2.42.0", "prom/prometheus:v2.42.0-rc1", "prom/prometheus:v2.44.0", "prom/prometheus:v2.45.0" }
```

etc

Reply:
Yeah I did that because they called out to the internet. But providing a canned list of available tags and testing against that would be better. I'd work on this and send an updated PR by the end of the day